cache/keytemplate/keytemplate_test.go

```diff
@@ -124,6 +127,62 @@ func TestEvaluate(t *testing.T) {
 			}
 		})
 	}
+
+	t.Run("Key with home file", func(t *testing.T) {
```

Comment:
NIT: it was a bit hard to understand what and how we test here, because the rest of the test code is about test setup. Maybe I would move the temp dir and file creation in the user home into a separate func. 
Something like:

```
t.Run("Key with home file", func(t *testing.T) {
	checksum, err := createTestFile("~/go-steputils-test/testfile)

	l := log.NewLogger()
	l.EnableDebugLog(true)
	l.Debugf("Tempfile: %s", tmpFile.Name())
	model := Model{
		envRepo: envRepository{},
		logger:  l,
		os:      "darwin",
		arch:    "arm64",
	}
	got, err := model.Evaluate(`gradle-cache-{{ checksum "~/go-steputils-test/*" }}`)
	if err != nil {
		t.Errorf("Evaluate() error = %v", err)
		return
	}

	want := fmt.Sprintf("gradle-cache-%x", checksum)
	if got != want {
		t.Errorf("Evaluate() got = %v, want %v", got, want)
	}
})
```